### PR TITLE
JAMES-3375 Restricted Email/query to personal mbx when no SHARE extension

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[ERROR] Failed to execute goal on project james-server-task-api: Could not resolve dependencies for project org.apache.james:james-server-task-api:jar:3.6.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus.mojo:animal-sniffer-annotations:pom:1.14 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.pom.part (No such file or directory) -> [Help 1]
```

network issue when downloading some JARs.

test this please

**Comment by chibenwa** [comment]
Compile time again

test this please

Otherwise Ill have a look tomorrow

**Comment by chibenwa** [comment]
test this please

Deployment

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 1]
```diff
@@ -155,6 +156,395 @@ trait EmailQueryMethodContract {
...
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+    awaitAtMostTenSeconds.untilAsserted { () =>
+      val response = `given`
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(request)
+      .when
+        .post.prettyPeek()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:194]: debug spotted

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 2]
```diff
@@ -1369,6 +1402,37 @@ void searchForMessageShouldPriorizeExclusionFromInclusion() throws Exception {
...
+                sessionFromDelegater);
+
+            MultimailboxesSearchQuery multiMailboxesQuery = MultimailboxesSearchQuery
+                .from(SearchQuery.matchAll())
+                .inNamespace(new PersonalNamespace(session))
+                .build();
+
+            assertThat(Flux.from(mailboxManager.search(multiMailboxesQuery, session, DEFAULT_MAXIMUM_LIMIT))
+                .collectList().block())
+                .isEmpty();
```
Comment (Arsnael) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java:1433]: indent, skip a line with block as well

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 3]
```diff
@@ -1255,6 +1257,37 @@ void searchForMessageShouldReturnMessagesFromMyDelegatedMailboxes() throws Excep
...
+                sessionFromDelegater);
+
+            MultimailboxesSearchQuery multiMailboxesQuery = MultimailboxesSearchQuery
+                .from(SearchQuery.matchAll())
+                .inNamespace(new AccessibleNamespace())
+                .build();
+
+            assertThat(Flux.from(mailboxManager.search(multiMailboxesQuery, session, DEFAULT_MAXIMUM_LIMIT))
+                .collectList().block())
+                .containsOnly(messageId);
```
Comment (Arsnael) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java:1288]: indent, skip a line with block as well

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 4]
```diff
@@ -155,6 +156,395 @@ trait EmailQueryMethodContract {
...
+         |      "filter": {"inMailbox": "${andreInboxId.serialize()}"}
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+      val response = `given`
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(request)
+      .when
+        .post.prettyPeek()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:319]: debug spotted

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
and rebase needed

**Comment by chibenwa** [comment]
(deployment tests)

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

(deployment tests)

**Review by nvduc91 — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

Deployment rests

**Comment by chibenwa** [comment]
test this please

Deployment

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**